unit/solvers/prop/bdd_expr.cpp

```diff
@@ -44,4 +44,23 @@ SCENARIO("bdd_expr", "[core][solver][prop][bdd_expr]")
       REQUIRE(bdd.as_expr() == bdd2.as_expr());
     }
   }
+
+  GIVEN("A bdd for (a&!b)")
+  {
+    const symbol_exprt a("a", bool_typet());
+    const symbol_exprt b("b", bool_typet());
+
+    const bddt bdd = bdd_expr_converter.from_expr(and_exprt(a, not_exprt(b)));
+
+    WHEN("It is converted to an exprt")
+    {
+      const exprt result = bdd_expr_converter.as_expr(bdd);
+      THEN("It is equal to the expression (a & !b) or (!b & a)")
+      {
+        const bool correct = result == and_exprt(a, not_exprt(b)) ||
+                             result == and_exprt(not_exprt(b), a);
+        REQUIRE(correct);
```

Comment:
I'd break this up into four `REQUIRE` statements to obtain more meaningful output in case of failure:
```
REQUIRE(result.id() == ID_and);
REQUIRE(result.operands().size() == 2);
REQUIRE(result.op0() == a || result.op1() == a);
REQUIRE(result.op0() == not_exprt(b) || result.op1() == not_exprt(b));
```